### PR TITLE
improve output readability of unit tests with test_base.h

### PR DESCRIPTION
**Pull request by rodolforg**
Although the programming interface didn't change,
test functions are not required to return a bool value anymore

### CODE REVIEW

**Comment by rodolforg** [comment]
For comparison, this is the current output printed by `test/node.cpp`:
<pre>$ ./_production/synfig-core/test/node
deleting_node_removes_it_as_parent_from_its_children:274
	 - expected 0, but got 1
<font color="#C01C28">synfig(144358) [00:33:04] error: deleting_node_removes_it_as_parent_from_its_children FAILED</font>
<font color="#C01C28">synfig(144358) [00:33:04] error: Test finished with 1 errors and 25 successful tests</font>
</pre>
And this is the output printed with this PR to same file:
<pre>$ ./_production/synfig-core/test/node
...................F......
<font color="#A2734C">synfig(124845) [00:19:57] warning: FAILURE:</font>
<font color="#A2734C"> deleting_node_removes_it_as_parent_from_its_children:236</font>
	 - expected 0, but got 1


========================================================================
<font color="#A2734C">synfig(124845) [00:19:57] warning: Statistics:</font>
<font color="#A2734C"> 1 tests failed.</font>
<font color="#A2734C"> 25 successful tests</font>
========================================================================
</pre>

PS: The difference in the line number is due to removal of lines `return false;` that aren't required anymore.

File: synfig-core/test/test_base.h [hunk 1]
```diff
@@ -22,77 +22,89 @@
...
 
-#include <iostream> // std::cerr
 #include <cstdlib> // std::abs
+#include <iostream> // std::cerr
+#include <sstream>
 
 #include <synfig/general.h> // synfig::error , synfig::info
 #include <synfig/vector.h>
 
+struct SynfigTestException
```
Comment (ice0) [inline synfig-core/test/test_base.h:32]: Maybe inherit it from `std::exception`?

Comment (rodolforg) [reply synfig-core/test/test_base.h]: Done

**Review by ice0 — APPROVED** [review]

**Comment by ice0** [comment]
Merged. Thank you!

**Pull request merged**